packages/analytics-js-common/src/types/LoadOptions.ts

```diff
@@ -138,19 +172,31 @@
   secureCookie?: boolean; // defaults to false.
   destSDKBaseURL?: string; // defaults to https://cdn.rudderlabs.com/latest/v3/modern/js-integrations
   pluginsSDKBaseURL?: string; // defaults to https://cdn.rudderlabs.com/latest/v3/modern/plugins
+  /**
+   * @deprecated The beacon functionality is automatically supported by the SDK in a different form.
+   * This option is no longer required.
+   */
   useBeacon?: boolean; // defaults to false.
+  /**
+   * @deprecated Use queueOptions instead
+   */
   beaconQueueOptions?: BeaconQueueOpts;
   destinationsQueueOptions?: DestinationsQueueOpts;
   anonymousIdOptions?: AnonymousIdOptions;
+  /**
+   * @deprecated Use storage.cookie.domain instead
+   */
   setCookieDomain?: string; // defaults to current domain.
+  /**
+   * @deprecated Use storage.cookie.sameSite instead
+   */
```

Comment:
Since we use same types for public-facing APIs and internal code base, they are actually deprecated for the users but we're providing backward compatibility.